File: FRONT/AmusementPark/src/app/models/users/user_dto.ts
export interface UserDto {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  publicDisplayName?: string | null;
  publicIdentifier?: string | null;
  isActivated: boolean;
  isBlocked: boolean;
  roles: string[];
  preferredLanguage: string;
  preferredMeasurementSystem?: string | null;
  avatarUrl: string;
  createdAt: string;
  updatedAt: string;
}

File: API/AmusementPark.WebAPI/Contracts/Users/UserDto.cs
using System;
using System.Collections.Generic;

namespace AmusementPark.WebAPI.Contracts.Users;

/// <summary>
/// Contrat HTTP de listing utilisateur.
/// </summary>
public sealed class UserDto
{
    public string Id { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public string? FirstName { get; set; }

    public string? LastName { get; set; }

    public string? Email { get; set; }

    public bool IsActivated { get; set; }

    public bool IsBlocked { get; set; }

    public string? PreferredLanguage { get; set; }

    public string? PreferredMeasurementSystem { get; set; }

    public List<UserRoleDto> Roles { get; set; } = new();


    public string? AvatarUrl { get; set; }
}
